docs/advanced.qmd

```diff
@@ -78,14 +81,60 @@ Use this when storing bundles on servers with public HTTP access or cloud storag
 
 ### Configuration Scripts
 
-Included configuration files can execute validation code to enforce constraints. Files named `_flow.py` are automatically included from parent directories, and receive `__flow_including_jobs__` containing all configs in the include chain.
-
-**Terminology:**
-
-- **Including config**: The file being loaded (e.g., `my_config.py`)
-- **Included config**: The `_flow.py` file(s) automatically inherited from parent directories
-
-The included config can inspect the including configs to validate or enforce constraints.
+When loading a configuration file, Flow expects the last statement to either be a `FlowJob` or a function that returns a `FlowJob`.
+Other than this requirement, the configuration file may execute arbitrary code.
+One example of functionality that can be implemented using this feature is validation code to enforce constraints.
+Instead of repeating this validation code in every Flow configuration file, the code could be placed in a _flow.py file that is auto included.
+In this section, first we provide a detailed example of a Flow include structure.
+Then we provide example configuration scripts that can be used to enforce constraints.
+
+#### Flow Includes Detailed Example
+
+Flow files may be included in two ways:
+- Files named `_flow.py` are automatically included from parent directories.
+  - When using the CLI to load a file, only _flow.py files in the parent directories of the configuration file provided are automatically included.
+  - When the API prepares a job that contains includes, only _flow.py files in the parent directories of the base_dir arg are automatically included.
+- Other configuration files can be explicitly included using the `includes` field in the `FlowJob`.
+
+Assume the following file structure:
+~/_flow.py
+~/a/_flow.py
+~/a/b/1flow.py
+~/a/b/2flow.py
+~/a/b/my_config.py
+~/a/c/_flow.py
+~/a/c/2_1flow.py
+
+And the following includes:
+- 2flow.py includes 2_1flow.py
+- my_config-py includes 1flow.py and 2flow.py
+
+~/a/c/_flow.py will not be loaded.
+Even though ~/a/c/2_1flow.py is loaded from the directory, Flow only automatically includes files in the same directory is my_config.py and its parents.
+
+The config files will be loaded and applied in the following order:
```

Comment:
I think it's worth clarifying that this order is triggered when `my_config.py` is being run or resolved.